src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java

```diff
@@ -29,112 +26,109 @@
-    private final ItemResponseMapper itemResponseMapper;
-    private final UserResponseMapper userResponseMapper;
+    private final BookingDAO bookingDAO;
+    private final UserDAO userDAO;
+    private final ItemDAO itemDAO;
+    private final BookingMapper bookingMapper;
+    private final ItemMapper itemMapper;
+    private final UserMapper userMapper;
 
     @Override
     @Transactional
-    public BookingResponseDto create(BookingRequestDto requestDto) {
+    public BookingOutputDTO create(BookingInputDTO inputDTO) {
 
-        UserResponseDto booker = validateUserById(requestDto.getBookerId());
-        ItemResponseDto item = validateItemById(requestDto.getItemId());
+        UserOutputDTO booker = validateUserById(inputDTO.getBookerId());
+        ItemShortOutputDTO item = validateItemById(inputDTO.getItemId());
         Long bookerId = booker.getId();
         Long itemId = item.getId();
 
-        if (itemDao.existsItemByIdAndOwner_Id(itemId, bookerId)) {
+        if (itemDAO.existsItemByIdAndOwner_Id(itemId, bookerId)) {
             throw NotFoundException.builder()
                     .message(String.format("The user with an ID - `%d` is creating item with an ID - `%d` and cannot booking it.", itemId, bookerId))
                     .build();
         }
 
-        requestDto.setStatus(WAITING);
-        ValidatorUtils.validate(requestDto, Marker.OnCreate.class);
-        validateDateTime(requestDto);
+        inputDTO.setStatus(WAITING);
+        validateDateTime(inputDTO);
 
-        BookingResponseDto responseDto = bookingResponseMapper.toDto(bookingDao.save(bookingRequestMapper.toEntity(requestDto)));
-        responseDto.setBooker(booker);
-        responseDto.setItem(item);
+        BookingOutputDTO outputDto = bookingMapper.toOutputDTO(bookingDAO.save(bookingMapper.inputDTOToEntity(inputDTO)));
+        outputDto.setBooker(booker);
+        outputDto.setItem(item);
 
-        return responseDto;
+        return outputDto;
     }
 
     @Override
-    @Transactional(readOnly = true)
-    public BookingResponseDto getById(Long userId, Long bookingId) {
+    public BookingOutputDTO getById(Long userId, Long bookingId) {
 
-        return bookingResponseMapper.toDto(
-                bookingDao.findBookingByIdAndBooker_IdOrIdAndItem_Owner_Id(bookingId, userId, bookingId, userId)
+        return bookingMapper.toOutputDTO(
+                bookingDAO.findBookingByIdAndBooker_IdOrIdAndItem_Owner_Id(bookingId, userId, bookingId, userId)
                         .orElseThrow(() -> NotFoundException.builder()
                                 .message(String.format("The booking with the ID - `%d` was not found.", bookingId))
                                 .build()));
     }
 
     @Override
     @Transactional
-    public BookingResponseDto approveBooking(Long ownerId, Long bookingId, boolean approved) {
+    public BookingOutputDTO approveBooking(Long ownerId, Long bookingId, boolean approved) {
```

Comment:
Добавьте проверку, что текущий пользователь (ownerId) действительно является владельцем элемента перед тем, как изменять статус бронирования. Это важно для поддержания целостности данных и предотвращения несанкционированных изменений.

Reply:
Логика проверки доступности элемента и его статуса должна быть вынесена в отдельный приватный метод, чтобы избежать дублирования кода и улучшить читаемость.

Reply:
Данная проверка уже присутствует.
На уровне сервисного слоя при подтверждении бронирования, обращение в DB запросным методом.
```suggestion
     if (!itemDAO.existsItemByIdAndOwner_Id(itemId, ownerId)) {
            throw NotFoundException.builder()
                    .message(String.format("The item with the ID - `%d` does not belong to the user with the ID - `%d`.", itemId, ownerId))
                    .build();
        }
```

Reply:
Проверки вынес в приватный метод.